sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -50,7 +57,14 @@ public class EventData
         ///   </code>
         /// </example>
         ///
-        public IDictionary<string, object> Properties { get; }
+        public IDictionary<string, object> Properties
+        {
+            get
```

Comment:
I had considered wrapping the `Interlocked` call in an `if (_properties == null)` block but, based on some microbenchmarking, it seems that the branch logic is more expensive than doing the interlocked operation always.  This definitely could be misleading, so thoughts would be appreciated.

```text

|                 Method |     Mean |    Error |   StdDev |    Gen 0 |    Gen 1 | Gen 2 | Allocated |
|----------------------- |---------:|---------:|---------:|---------:|---------:|------:|----------:|
|                UseSafe | 258.0 μs |  5.05 μs |  8.71 μs | 144.5313 |  65.4297 |     - |   1.16 MB |
|           UseSafenNoIf | 229.4 μs |  4.43 μs |  4.92 μs | 144.5313 |  65.6738 |     - |   1.16 MB |
|          UseDictionary | 439.1 μs |  5.58 μs |  5.22 μs | 240.2344 | 114.2578 |     - |   1.92 MB |
|       UseSafeWithProps | 924.9 μs |  7.47 μs |  6.62 μs | 335.9375 | 163.0859 |     - |   2.68 MB |
|   UseSafeNoIfWithProps | 888.7 μs |  8.46 μs |  7.91 μs | 335.9375 | 164.0625 |     - |   2.68 MB |
| UseDictionaryWithProps | 850.4 μs | 14.44 μs | 13.51 μs | 335.9375 | 163.0859 |     - |   2.68 MB |
```

`UseSafe` == The interlocked approach with an `if` wrapper
`UseSafeNoIf` == The interlocked approach with no `if` wrapper
`UseDictionary` == The current eager allocation of the dictionary (baseline)

The first set of operations just creates an bunch of `EventData` instances with the same body and no property use.  The second set creates a bunch of `EventData` instances and sets the same set of properties on each.